oak_functions_containers_app/src/lib.rs

```diff
@@ -136,7 +142,7 @@ impl OakFunctions for OakFunctionsContainersService {
         &self,
         request: tonic::Request<InvokeRequest>,
     ) -> Result<tonic::Response<InvokeResponse>, tonic::Status> {
-        let encryption_key_provider = self.encryption_key_provider.clone();
+        let encryption_key_provider = self.orchestrator_client.clone();
```

Comment:
So I think we would be able to use an `AsyncEncryptionHandler` below -- or does that not work?

Reply:
No, because the code expects an `EncryptionHandler` and `invoke()` is not async.

Reply:
Sorry still not sure I follow. If we used `AsyncEncryptionHandler`, then invoke would become async, and we can await it since the handler is already async.

Reply:
also not sure I follow

Reply:
and I'm not sure I follow -- where exactly I should be able to use the AsyncEncryptionHandler?

Reply:
Oh. There. :)

Yes, much cleaner now.